client/src/Shell/DistributeSong/AddAlbum/LoadAlbumDialog/index.tsx

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useState } from 'react';
+import {
+  Dialog, DialogTitle, FormControl, Grid, IconButton, InputAdornment, InputLabel, MenuItem, Paper,
+  Select, Stack, TextField
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import { includes, sortBy } from 'lodash';
+import SearchIcon from '@mui/icons-material/Search';
+import { Album } from '../../../../types/Album';
+import AlbumCardCell from '../AlbumCardCell';
+import useFetch from '../../../../hooks/use-fetch';
+
+interface LoadAlbumDialogProps {
+  open: boolean,
+  handleClose: () => void
+}
+
+const initialSort = 'name';
+
+const LoadAlbumDialog = ({ open, handleClose }: LoadAlbumDialogProps) => {
+  const [albumsToDisplay, setAlbumsToDisplay] = useState<Album[]>([]);
+  const [sortByCriteria, setSortByCriteria] = useState(initialSort);
+  const [albums] = useFetch<Album>('albums');
+
+  const handleSortChange = (value: string) => {
+    setSortByCriteria(value);
+    setAlbumsToDisplay(sortBy(albums, [value]));
+  };
+
+  const handleSearchChange = (value: string) => {
+    setAlbumsToDisplay(value
+      ? albums.filter(x => includes(x.name.toLowerCase(), value.toLowerCase()))
+      : albums);
+  };
+
+  useEffect(() => {
+    handleSortChange(initialSort);
+    handleSearchChange('');
+  }, [albums]);
+
+  return (
+    <Dialog open={open} onClose={handleClose} maxWidth='md' fullWidth>
+      <Grid container direction='column' sx={{ p: 2, m: 0 }}>
+        <Grid container justifyContent='space-between'>
+          <DialogTitle>טעינת אלבום</DialogTitle>
+          <IconButton onClick={handleClose}>
+            <CloseIcon />
+          </IconButton>
+        </Grid>
+        <Stack direction='row' spacing={2} sx={{ margin: 2 }}>
+          <TextField
+            label='חפש לפי שם'
```

Comment:
sperate the search bar by name , by criteria and the list of albums to some components